dev-utils/docker-compose.yml

```diff
@@ -74,46 +89,71 @@ services:
       test: ["CMD", "curl", "--write-out", "'HTTP %{http_code}'", "--silent", "--output", "/dev/null", "http://kibana:5601/"]
       retries: 10
       interval: 10s
+    mem_limit: 2000m
+#    cpus: 0.5
     depends_on:
       elasticsearch:
         condition: service_healthy
     labels:
       - "co.elatic.apm.stack-version=${STACK_VERSION}"
+    networks:
+      apm:
+        aliases:
+          - kibana
 
-  apm-server-dev:
-    build:
-      context: ./apm-server-dev
-      args:
-        APM_SERVER_REPO: ${APM_SERVER_REPO}
-        APM_SERVER_BRANCH: ${APM_SERVER_BRANCH}
-    ports:
-      - "127.0.0.1:${APM_SERVER_PORT:-8200}:8200"
+  wait-for-apm-server:
+    container_name: wait-for-apm-server
+    image: alpine
     command: >
-      ./apm-server -e
-        -E apm-server.rum.enabled=true
-        -E apm-server.rum.rate_limit=100000
-        -E apm-server.host=0.0.0.0:8200
-        -E apm-server.read_timeout=1m
-        -E apm-server.shutdown_timeout=2m
-        -E apm-server.write_timeout=1m
-        -E setup.template.settings.index.number_of_replicas=0
-        -E output.elasticsearch.hosts=http://elasticsearch:9200
-    cap_drop:
-      - ALL
-    cap_add:
-      - CHOWN
-      - DAC_OVERRIDE
-      - SETGID
-      - SETUID
-    logging:
-      driver: 'json-file'
-      options:
-          max-size: '2m'
-          max-file: '5'
-    depends_on:
-      elasticsearch:
-        condition: service_healthy
+      /bin/sh -c exit 0
     healthcheck:
-      test: ["CMD", "curl", "--write-out", "'HTTP %{http_code}'", "--silent", "--output", "/dev/null", "http://apm-server:8200/"]
+      test: ["CMD", "/bin/sh", "-c", "exit 0"]
       retries: 10
-      interval: 10s
\ No newline at end of file
+      interval: 10s
+    depends_on:
+      apm-server:
+        condition: service_healthy
+    networks:
+      apm:
+        aliases:
+          - wait-for-apm-server
+
+  node-puppeteer:
+    container_name: node-puppeteer
+    image: docker.elastic.co/observability-ci/node-puppeteer:8
+    environment:
+      - HOME=/app
+      - SCOPE=${SCOPE}
+      - STACK_VERSION=${STACK_VERSION}
+      - APM_SERVER_URL=${APM_SERVER_URL}
+      - BUILD_NUMBER=${BUILD_NUMBER}
+      - BRANCH_NAME=${BRANCH_NAME}
+      - JENKINS_URL=${JENKINS_URL}
+      - SAUCE_USERNAME=${SAUCE_USERNAME}
+      - SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY}
+      - MODE=${MODE}
```

Comment:
@kuisathaverat I have a PR that runs the sauce tests only when the mode is set to saucelabs, Lets not set the `MODE=saucelabs` for Jenkins till we get a new account. 

The PR is here - https://github.com/elastic/apm-agent-rum-js/pull/312

Happy to merge this PR once we can change this behaviour. I can also push to this branch.

Reply:
cool, because I've tested 'build:main build:e2e test:node test:integration' and always try to start the connection to SuceLab because it does not uses the same functions from `devutils/testutils.js` nor `devutils/karma.js`, it calls directly `devutils/testutils.runSauceConnect` and ignore the `devutils/karma.js` configuration

Reply:
yeah mode was set to saucelabs before and that was the reason. Its fixed in other PR. I will try to merge that today and rebase this one to be merged to master.

Reply:
@kuisathaverat Can you rebase with master and give it a go? Or else I can also take care of those changes. Let me know if you need any help. 

For now we can disable all the sauce labs tests on Jenkins by not setting the `MODE=saucelabs` flag.

Reply:
As far as I see the `MODE=saucelabs` is managed with an input boolean parameter which it's disabled now:
- https://github.com/elastic/apm-agent-rum-js/pull/180/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R42

I'll proceed with the rebase with master and push those changes now

Reply:
Cool, do you want to rebase the PR? The tests should pass once we do that.

Reply:
it does not need to rebase, every build merge master changes before to build

Reply:
awesome, so just a re trigger should do the job. Thanks!